src/platform_impl/web/web_sys/mod.rs

```diff
@@ -52,38 +53,13 @@ impl WindowExtWebSys for Window {
     }
 }
 
-pub fn window_size(window: &web_sys::Window) -> LogicalSize<f64> {
-    let width = window
-        .inner_width()
-        .expect("Failed to get width")
-        .as_f64()
-        .expect("Failed to get width as f64");
-    let height = window
-        .inner_height()
-        .expect("Failed to get height")
-        .as_f64()
-        .expect("Failed to get height as f64");
-
-    LogicalSize { width, height }
-}
-
 pub fn scale_factor(window: &web_sys::Window) -> f64 {
     window.device_pixel_ratio()
 }
 
-pub fn set_canvas_size(canvas: &Canvas, new_size: Size) {
-    let scale_factor = scale_factor(canvas.window());
-
-    let physical_size = new_size.to_physical(scale_factor);
-    canvas.size().set(physical_size);
-
-    let logical_size = new_size.to_logical::<f64>(scale_factor);
-    set_canvas_style_property(canvas.raw(), "width", &format!("{}px", logical_size.width));
-    set_canvas_style_property(
-        canvas.raw(),
-        "height",
-        &format!("{}px", logical_size.height),
-    );
+pub fn set_canvas_size(raw: &HtmlCanvasElement, new_size: LogicalSize<f64>) {
+    set_canvas_style_property(raw, "width", &format!("{}px", new_size.width));
+    set_canvas_style_property(raw, "height", &format!("{}px", new_size.height));
```

Comment:
Why'd you get rid of all the logic for dealing with padding?

Reply:
I just reverted the last commit, I will re-add that logic, keep in mind this is still a draft!
Don't let me discourage you from leaving any comments though, your input is appreciated :heart:!

Reply:
Added back in https://github.com/rust-windowing/winit/pull/2859/commits/4a9b1a5612f461366a19aef076af0d48761b8eff.